docs/content/TopNQuery.md

```diff
@@ -211,4 +211,7 @@ Users who can tolerate *approximate rank* topN over a dimension with greater tha
     "queryType": "topN",
     "threshold": 2
 }
-```
\ No newline at end of file
+```
+
+### DimExtractionFn
```

Comment:
I can see it falling out in one of two ways:

1) It's a property of the namespace.  If the namespace is going to do a strict 1-to-1 mapping then it can be done on the broker without paying the cost on any historicals
2) It is done via a nested query, where you do the topN and then nest it in a select that applies the namespace

This is a more "generic" form and requires that nested queries actually work with more than just groupBy.  It also completely externalizes the choice, which can be both good (simplifies our code) and bad (makes external peeps do more shiz).  The primary con, I believe, is that it requires us to do work to make nested queries work more better before we can make further progress on this.

All of that said, I think we definitely need the ability to do lookups on the broker as well to support the filter case.  For filters that are on a namespaced dim extraction, I think we want the broker to do a single lookup and rewrite it into a large OR clause instead of doing the lookup on each historical.